src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java

```diff
@@ -496,6 +496,47 @@ public static int typeDeclarationNameMatchingCount(String patternClass,
         return result;
     }
 
+    /**
+     * Calculates and returns the type declaration matching count when
+     * {@code typeDeclarationToBeMatched} is considered to be the superclass of an anonymous inner
+     * class.
+     *
+     * <p>
+     * For example, if the pattern class is {@code Main.ClassOne} and the class to be matched is
+     * {@code Main.ClassOne.ClassTwo.ClassThree}, then the matching count would be calculated by
+     * comparing the characters at each position, and updating the count whenever a '.'
+     * is encountered.
+     * This is necessary because pattern class can include anonymous inner classes, unlike regular
+     * inheritance where nested classes cannot be extended.
+     * </p>
+     *
+     * @param patternTypeDeclaration type declaration to match against
+     * @param typeDeclarationToBeMatched type declaration to be matched
+     * @return the type declaration matching count
+     */
+    public static int getAnonSuperTypeMatchingCount(String patternTypeDeclaration,
```

Comment:
This method looks very specific to Check, please consider to move it to Check.
We discourage until methods, and ok only if there is already few Checks are going to use it.

Reply:
Moved `getAnonSuperTypeMatchingCount` to `UnusedLocalVariableCheck`, and also `typeDeclarationNameMatchingCount` to `FinalClassCheck` as it's the only class that uses this method now.